lib/archethic/pub_sub.ex

```diff
@@ -67,6 +68,14 @@ defmodule ArchEthic.PubSub do
     dispatch(:new_oracle_data, {:new_oracle_data, data})
   end
 
+  @doc """
+  Notify next summary time beacon chain to the subscribers
+  """
+  def notify_next_summary_time(date = %DateTime{}) do
+    next_summary_time = BeaconChain.next_summary_date(date)
```

Comment:
Again several times the same thing, because from your function above you pass the next summary time to this function, so you will get again the same or the next one depending on the seconds/microseconds.
This function should just straightforward: given a time, it will notify all subscribers.

Reply:
so, changed notify fn to this

  def notify_next_summary_time(date = %DateTime{}) do
    dispatch(:next_summary_time, {:next_summary_time, date})
  end